pkg/kubelet/images/image_manager.go

```diff
@@ -164,6 +164,10 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, pod *v1.Pod, conta
 			msg := fmt.Sprintf("image pull failed for %s because the registry is unavailable.", container.Image)
 			return "", msg, imagePullResult.err
 		}
+		if imagePullResult.err.Error() == ErrInvalidSignature.Error() {
```

Comment:
It's potentially worth pulling these `if`'s into a new `func messageForCRIPullError(container, err) string` - adding more cases here is a bit unweildy as the function is already nearly a hundred lines long

Reply:
Yep, moved it into a separate function and added unit tests for it.